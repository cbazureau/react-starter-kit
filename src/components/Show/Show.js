import React from 'react';
import './Show.scss';
import Loader from '../../components/Loader/Loader';

const Show = ({ show }) => (show ?
  <div className="Show">
    <h1 className="Show__name">{show.name}</h1>
    <div className="Show__img">
      <img src={show.image.medium} alt={`Poster ${show.name}`} />
    </div>
    <div className="Show__desc">
      <p>{show.summary.replace(/(<([^>]+)>)/ig, '')}</p>
    </div>
  </div> : <Loader />
);

Show.propTypes = {
  show: React.PropTypes.shape({
    id: React.PropTypes.number.isRequired,
    name: React.PropTypes.string.isRequired,
    summary: React.PropTypes.string.isRequired,
    image: React.PropTypes.shape({
      medium: React.PropTypes.string.isRequired,
    }),
  }),
};

export default Show;
