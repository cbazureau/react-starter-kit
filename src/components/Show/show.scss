@import '../../constants.scss';

.Show {
  .Show__img {
    float:left;
    width: 18rem;
    img {width:18rem}
  }
  .Show__desc {
    float:right;
    width: calc(100% - 200px);
  }
  &:after {
    display: table;
    content: "";
    clear: both;
  }
}

