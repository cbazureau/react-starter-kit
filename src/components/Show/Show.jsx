import React from 'react'
import './show.scss'
import Loader from '../../components/Loader/Loader'

const Show = ({ show }) => (show ?
  <div className="show">
    <h1 className="show-name">{show.name}</h1>
    <div className="show-img">
      <img src={show.image.medium} alt={`Poster ${show.name}`} />
    </div>
    <div className="show-desc" dangerouslySetInnerHTML={{ __html: show.summary }} />
  </div> : <Loader />
)

Show.propTypes = {
  show: React.PropTypes.shape({
    id: React.PropTypes.number.isRequired,
    name: React.PropTypes.string.isRequired,
    summary: React.PropTypes.string.isRequired,
    image: React.PropTypes.shape({
      medium: React.PropTypes.string.isRequired
    })
  })
}

export default Show
