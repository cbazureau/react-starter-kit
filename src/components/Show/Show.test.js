import React from 'react';
import renderer from 'react-test-renderer';
import Show from './Show';

const mock = {
  id: 12,
  name: 'Test Name',
  summary: '<p>Lorem ipsum</p>',
  image: {
    medium: 'http://img',
  },
};

describe('Show', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Show show={mock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
