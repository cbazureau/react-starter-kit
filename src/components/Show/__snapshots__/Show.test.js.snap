// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show should render correctly 1`] = `
<div
  className="Show"
>
  <h1
    className="Show__name"
  >
    Test Name
  </h1>
  <div
    className="Show__img"
  >
    <img
      alt="Poster Test Name"
      src="http://img"
    />
  </div>
  <div
    className="Show__desc"
  >
    <p>
      Lorem ipsum
    </p>
  </div>
</div>
`;
